src/lib.rs

```diff
@@ -395,27 +524,20 @@ impl<T> Bus<T> {
             if fence_read == self.expected(fence) {
                 // yes! go ahead and write!
                 break;
-            } else if block {
+            }
+            match P::maybe_park(&mut prepark, |current| {
                 // no, so block by parking and telling readers to notify on last read
                 self.state.ring[fence]
                     .waiting
-                    .replace(Some(Box::new(thread::current())), atomic::Ordering::Relaxed);
+                    .replace(Some(Box::new(current)), atomic::Ordering::Relaxed);
```

Comment:
I'm really sad about this `Box`, but unfortunately I think it's necessary. We could perhaps at least hoist it outside the `loop`..?

Reply:
`AtomicOption` requires the `Box` though.